app/scripts/metamask-controller.js

```diff
@@ -114,6 +115,19 @@ export default class MetamaskController extends EventEmitter {
       migrateAddressBookState: this.migrateAddressBookState.bind(this),
     })
 
+    // This depends on preferences controller state
+    this.trackSegmentEvent = getTrackSegmentEvent(
+      () => this.preferencesController.getParticipateInMetaMetrics(),
+      () => this.platform.getVersion(),
```

Comment:
Nit: The version can't change at runtime, so it can be passed in as a string rather than a function

Reply:
e55b6d7